src/dashboard/src/components/file/views.py

```diff
@@ -0,0 +1,210 @@
+# Standard library, alphabetical by import source
+import json
+import logging
+import os
+import sys
+
+# Django Core, alphabetical by import source
+
+# External dependencies, alphabetical
+import requests
+
+# This project, alphabetical by import source
+from components import helpers
+from main import models
+
+sys.path.append("/usr/lib/archivematica/archivematicaCommon")
+import elasticSearchFunctions
+import storageService as storage_service
+
+logger = logging.getLogger('archivematica.dashboard')
+
+def transfer_file_tags(request, uuid):
+    """
+    Fetch, set or remove tags from a record.
+    All methods return 404 if a record can't be retrieved for the given UUID, and 400 if more than one file was found for a given UUID.
+
+    Accepted methods and parameters:
+        GET: requires a file UUID.
+            Returns a JSON-encoded list of the file's tags on success.
+        PUT: requires a file UUID, and document body must be a JSON-encoded list.
+            Replaces the list of tags in the record with the provided list.
+            Returns {"success": true} on success.
+            Returns 400 if no JSON document is provided in the request body, if the body can't be decoded, or if the body is any JSON object other than a list.
+        DELETE: requiest a file UUID.
+            Deletes all tags for the given file.
+    """
+    if request.method == 'GET':
+        try:
+            tags = elasticSearchFunctions.connect_and_get_file_tags(uuid)
+        except elasticSearchFunctions.ElasticsearchError as e:
+            response = {
+                'success': False,
+                'message': str(e),
+            }
+            if isinstance(e, elasticSearchFunctions.EmptySearchResultError):
+                status_code = 404
+            else:
+                status_code = 400
+            return helpers.json_response(response, status_code=status_code)
+
+        return helpers.json_response(tags)
+    elif request.method == 'PUT':
+        try:
+            tags = json.load(request)
```

Comment:
Should this be `json.loads(request.body)`?  Or does `request` expose a file-like object of the body for `json.load`?